### PR TITLE
[inference] add type hints and named arguments

### PR DESCRIPTION
**Pull request by flozi00**
While reading the code i just added type hints and named arguments for function calls to improve readability and documentation

This is just for the inference folder at the moment

### CODE REVIEW

File: inference/server/export.py [hunk 1]
```diff
@@ -170,7 +170,7 @@ async def fetch_eligible_chats(session_generator, filters: list[Any]) -> list[Db
     filters.append(DbChat.allow_data_use)
     async with session_generator() as session:
         query = (
-            sqlmodel.select(DbChat)
+            sqlmodel.select(entity_0=DbChat)
```
Comment (olliestanley) [inline inference/server/export.py:173]: I think some of the named arguments are definitely helpful, but others (like this one) are not useful and imo detract from readability

Maybe we could have a slightly more case-by-case approach with some of the less useful ones taken out

**Review by olliestanley — CHANGES_REQUESTED** [review]

File: inference/server/export.py (hunk 1 above)
Comment (flozi00) [reply inference/server/export.py:173]: removing for all functions call where only one argument is passed ?

Comment (olliestanley) [reply inference/server/export.py]: Yeah, something like that would make sense to me

**Comment by melvinebenezer** [comment]
Thanks. It's good but can we make smaller changes ? Not all named arguments help in readability.

**Comment by olliestanley** [comment]
> Could you point out some examples? I hope I removed all the single argument changes, so only multi arg calls are named

There are still many single argument changes. I would also advocate for removal of cases where the parameter name is the same as the value we are passing in anyway. For example, reading `anonymizer=anonymizer`, or `session=session` is more of a distraction and doesn't really contribute to clarity, imo.

**Pull request closed without merging**